src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -0,0 +1,49 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@inline function check_bounds(u, mesh::AbstractMesh{2}, equations, solver, cache,
+                              limiter::SubcellLimiterIDP,
+                              time, iter, output_directory, save_errors)
+    (; positivity) = solver.volume_integral.limiter
+    (; variable_bounds) = limiter.cache.subcell_limiter_coefficients
+    (; idp_bounds_delta) = limiter.cache
+
+    if positivity
+        for v in limiter.positivity_variables_cons
+            key = Symbol(string(v), "_min")
+            deviation = idp_bounds_delta[key]
```

Comment:
```suggestion
            deviation = idp_bounds_delta[key]
            deviation[1] = zero(eltype(idp_bounds_delta[key][1]))
```
shouldn't you reset the "local" deviation before you compute it here? otherwise if `save_errors=false` it will never be reset. Or am I missing something?

Reply:
No, I don't think so. The idea is to only reset the first entry after it is exported to the txt file. Then, we miss no deviations in the txt file.
That's why I reset it [here](https://github.com/trixi-framework/Trixi.jl/pull/1672#discussion_r1366904054).

If `save_errors==false`, the first entry isn't used at all.
Or do I miss something?

Reply:
You're right.